src/datasource.ts

```diff
@@ -1,112 +1,135 @@
-      return this.$q.when({data: []});
-    }
-
-    if (this.templateSrv.getAdhocFilters) {
-      query.adhocFilters = this.templateSrv.getAdhocFilters(this.name);
-    } else {
-      query.adhocFilters = [];
-    }
-
-    return this.doRequest({
-      url: this.url + '/query',
-      data: query,
-      method: 'POST'
+  async query(options: DataQueryRequest<MetricQQuery>): Promise<DataQueryResponse> {
+    let query = this.buildQueryParameters(options);
+
+    query.targets = query.targets.filter((t: MetricQQuery) => !t.hide);
+
+    const data = await this.doRequest(this.url + '/query', 'POST', query).then((response: { data: any[] }) => {
+      return response.data.map((result: any) => {
+        const frame = new MutableDataFrame({
+          refId: result.refId,
+          name: result.target,
+          fields: [
+            { name: 'Time', type: FieldType.time },
+            { name: 'Value', type: FieldType.number },
+          ],
+          meta: {
+            stats: [
+              {
+                displayName: 'time-measurements-db',
+                value: result.time_measurements.db,
+              },
+              {
+                displayName: 'time-measurements-http',
+                value: result.time_measurements.http,
+              },
+            ],
+          },
+        });
+        result.datapoints.forEach((point: any) => {
+          frame.appendRow([point[1], point[0]]);
+        });
+
+        return frame;
+      });
     });
-  }
 
-  annotationQuery(options) {
-    throw new Error("Annotation Support not implemented yet.");
+    return { data };
   }
 
-  metricFindQuery(query: string) {
-    var interpolated = {
-      target: this.templateSrv.replace(query, null, 'regex')
+  async metricFindQuery(query: string): Promise<MetricFindValue[]> {
+    const interpolated = {
+      target: getTemplateSrv().replace(query, undefined, 'regex'),
     };
 
-    return this.doRequest({
-      url: this.url + '/search',
-      data: interpolated,
-      method: 'POST',
-    }).then(this.mapToTextValue);
+    return this.doRequest(this.url + '/search', 'POST', interpolated).then(this.mapToTextValue);
   }
 
-  testDatasource() {
-    return this.doRequest({
-      url: this.url + '/',
-      method: 'GET',
-    }).then(response => {
-      if (response.status === 200) {
-        return { status: "success", message: "Data source is working", title: "Success" };
-      }
-    });
+  async testDatasource() {
+    return getBackendSrv()
+      .datasourceRequest({
+        url: this.url + '/',
+        method: 'GET',
+      })
+      .then((response) => {
+        if (response.status === 200) {
+          return { status: 'success', message: 'Data source is working' };
+        } else {
+          return {
+            status: 'failed',
+            message: response.status,
+          };
+        }
+      });
   }
 
-  mapToTextValue(result) {
+  mapToTextValue(result: any): MetricFindValue[] {
```

Comment:
Replace with `mapToLabelValue`

Reply:
this function is used in `metricFindQuery` which requires `MetricFindValue` as return type